ocaml-lsp-server/src/custom_requests/req_polarity_search.ml

```diff
@@ -0,0 +1,124 @@
+open Import
+module TextDocumentPositionParams = Lsp.Types.TextDocumentPositionParams
+
+let meth = "ocamllsp/polaritySearch"
+let capability = "handlePolaritySearch", `Bool true
+
+let completion_kind kind : CompletionItemKind.t =
+  match kind with
+  | `Value -> Value
+  | `Variant -> EnumMember
+  | `Label -> Field
+  | `Module -> Module
+  | `Modtype -> Interface
+  | `MethodCall -> Method
+  | `Keyword -> Keyword
+  | `Constructor -> Constructor
+  | `Type -> TypeParameter
+;;
+
+let kind_of_completion_kind (kind : CompletionItemKind.t) =
+  match kind with
+  | Value -> `Value
+  | EnumMember -> `Variant
+  | Field -> `Label
+  | Module -> `Module
+  | Interface -> `Modtype
+  | Method -> `MethodCall
+  | Keyword -> `Keyword
+  | Constructor -> `Constructor
+  | TypeParameter -> `Type
+  | _ -> failwith "Unknown kind"
+;;
```

Comment:
There is already https://github.com/ocaml/ocaml-lsp/blob/3dc1a6633bfe2ebd34d7aa38f008c8c28300ee55/ocaml-lsp-server/src/compl.ml#L13 and since `merlin-search` always returns a kind of `Value`, I am not sure that having a full completion make sense here. WDYT?

Reply:
And would it be useful to allow to have the documentation in the "info" field ?

Reply:
Yes documentation can be great from the info field. Should Merlin find and return the documentation automatically in the results?

Reply:
Yes it could be useful but it need to change the polarity search function, make a new release of merlin, and affect this PR. It seems a lot of work for an hard-to-use feature ;)?

Reply:
If this feature is hard-to-use - essentially unusable until there's a working client, what's the rush to release it? Might as well wait until you adjust the API to make it work well while there's no clients relying on it.

Reply:
I could open an issue on Merlin linking to this comment thread, and we can also subsequently implement all the updates/enhancements in future PR's no?

Reply:
It would make sense if this work was already in a usable state and could provide some value to users while enhancements are made, but thus far, it's not really the case. Moreover, if the API in merlin changes and this feature is released, you will need to add constraints to already released versions. Sometimes, this isn't done in time, so users end up experiencing build failures.

On the other hand, if it stays as a PR, you're free to iterate on the feature in merlin, lsp, and the client all at the same time without worrying about backwards compat or versioning. Usually, the cost of that would be potential merge conflicts. But I see that the way you've written this feature in a fairly isolated manner, so the possibility of merge conflicts is low.

Ultimately, it's not forbidden to merge WIP as sometimes the cost of merge conflicts can be rather high. Evolving 3 concurrent code bases in a way that is user conscious is also not free though. My experience hints that in this particular case, the cost of the latter will outweigh the former.

Reply:
I think that the `Polarity search feature` is used by merlin User, in it's current state, but I think that the feature will not change in the future (and seems the name of the command is very precise : `polarity_search`), for me, it can be useful and not "to be deprecated". So for me, the command will probably not requiring any maintenance in the future.